include/codec.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API header file for Audio Codec
+ *
+ * This file contains the Audio Codec APIs
+ */
+
+#ifndef __CODEC_H__
```

Comment:
audio_codec.h maybe? (or anything that tells it's oriented towards audio. As there are also video codecs)

all the structs/enums defined will need to be changed accordingly.

Reply:
Changed.

Reply:
Since I now have another PR #8572, is it better to put the audio related APIs under `include/audio' as `include/audio/codec.h` and `include/audio/dmic.h`
Thoughts?